src/myjarvis/domain/services/chat_limits_service.py

```
@@ -0,0 +1,67 @@
+
+class MaxMessagesLimitStrategy(ChatLimitStrategy):
+    priority = 10
+
+    def apply(
+        self,
+        messages: List[Message],
+        limits: ChatLimits,
+    ) -> List[Message]:
+        if limits.max_messages is None or len(messages) <= limits.max_messages:
+            return messages
+        return messages[-limits.max_messages :]
```

Flake8 (7.2.0)

[error] whitespace before ':'

(E203)